src/query/Index.hpp

```diff
@@ -40,23 +41,32 @@ class Index {
   ~Index();
 
   /// Get n number of closest vertices to the given vertex
-  VertexMatch getClosestVertex(const mesh::Vertex &sourceVertex);
-
-  /// Get n number of closest vertices to the given vertex
+  VertexMatch              getClosestVertex(const mesh::Vertex &sourceVertex);
+  VertexMatch              getClosestVertex(const Eigen::VectorXd &sourceCoord);
   std::vector<VertexMatch> getClosestVertices(const mesh::Vertex &sourceVertex, int n);
+  std::vector<VertexMatch> getClosestVertices(const Eigen::VectorXd &sourceCoord, int n);
 
   /// Get n number of closest edges to the given vertex
+  EdgeMatch              getClosestEdge(const Eigen::VectorXd &sourceCoord);
   std::vector<EdgeMatch> getClosestEdges(const mesh::Vertex &sourcesVertex, int n);
+  std::vector<EdgeMatch> getClosestEdges(const Eigen::VectorXd &sourceCoord, int n);
 
   /// Get n number of closest triangles to the given vertex
+  TriangleMatch              getClosestTriangle(const Eigen::VectorXd &sourceCoord);
   std::vector<TriangleMatch> getClosestTriangles(const mesh::Vertex &sourceVertex, int n);
+  std::vector<TriangleMatch> getClosestTriangles(const Eigen::VectorXd &sourceCoord, int n);
 
   /// Return all the vertices inside the box formed by vertex and radius
   std::vector<size_t> getVerticesInsideBox(const mesh::Vertex &centerVertex, double radius);
 
   /// Return all the vertices inside a bounding box
   std::vector<size_t> getVerticesInsideBox(const mesh::BoundingBox &bb);
 
+  /// Temporary before other refactor, return Interpolation elements and the distance
+  std::pair<InterpolationElements, GenericMatch> findVertexProjectionElements(const mesh::Vertex &sourceVertex, int n);
+  std::pair<InterpolationElements, GenericMatch> findEdgeProjectionElements(const mesh::Vertex &sourceVertex, int n);
+  std::pair<InterpolationElements, GenericMatch> findTriangleProjectionElements(const mesh::Vertex &sourceVertex, int n);
```

Comment:
You are right regarding the `GenericMatch` it really is pointless.
Lets use only the distance to the matched primitive `std::pair<InterpolationElements, double>` for now. If we ever need the type of the matched primitive, we should add an enum to the GenericMatch which identifies the primitive it matched against.